map_server/src/map_saver.cpp

```diff
@@ -44,23 +44,37 @@ class MapGenerator
 {
 
   public:
-    MapGenerator(const std::string& mapname, int threshold_occupied, int threshold_free)
-      : mapname_(mapname), saved_map_(false), threshold_occupied_(threshold_occupied), threshold_free_(threshold_free)
+    MapGenerator(const std::string& mapname1, const std::string& mapname2, int threshold_occupied, int threshold_free)
+      : mapname1_(mapname1), mapname2_(mapname2), saved_map1_(false), saved_map2_(false), threshold_occupied_(threshold_occupied), threshold_free_(threshold_free)
     {
       ros::NodeHandle n;
-      ROS_INFO("Waiting for the map");
-      map_sub_ = n.subscribe("map", 1, &MapGenerator::mapCallback, this);
+      ROS_INFO("Waiting for the maps");
+
+      map_sub1_ = n.subscribe("map", 1, &MapGenerator::mapCallback1, this);
+      map_sub2_ = n.subscribe("map_hi_intensity", 1, &MapGenerator::mapCallback2, this);
```

Comment:
? does this work? i thought we dont need to change a source code if the remapping works as expected in [this PR](https://github.com/LexxPluss/LexxAuto/pull/3295/files). am i wrong?
![Screenshot from 2024-06-10 15-20-55](https://github.com/LexxPluss/navigation/assets/54845972/ada246af-ed5c-4652-8c65-03086a5b1ac6)

Reply:
Yes, it works. I have tested in on 63002.
Please check the comments [here](https://lexxpluss.atlassian.net/browse/AMRCS-293?focusedCommentId=21663).